operators/src/processing/temporal_raster_aggregation.rs

```diff
@@ -391,6 +478,93 @@ where
     )
 }
 
+fn first_tile_fold_fn<T>(
+    acc: TemporalRasterAggregationTileAccu<T>,
+    tile: RasterTile2D<T>,
+) -> TemporalRasterAggregationTileAccu<T>
+where
+    T: Pixel,
+{
+    if acc.initial_state {
+        let mut next_accu = tile;
+        next_accu.time = acc.accu_tile.time;
+
+        TemporalRasterAggregationTileAccu {
+            accu_tile: next_accu,
+            initial_state: false,
+        }
+    } else {
+        acc
+    }
+}
+
+pub fn first_tile_fold_future<T>(
+    accu: TemporalRasterAggregationTileAccu<T>,
+    tile: RasterTile2D<T>,
+) -> impl Future<Output = Result<TemporalRasterAggregationTileAccu<T>>>
+where
+    T: Pixel,
+{
+    tokio::task::spawn_blocking(|| first_tile_fold_fn(accu, tile)).then(async move |x| match x {
+        Ok(r) => Ok(r),
+        Err(e) => Err(e.into()),
+    })
+}
+
+#[allow(clippy::needless_pass_by_value)]
+fn last_tile_fold_fn<T>(
+    acc: TemporalRasterAggregationTileAccu<T>,
+    tile: RasterTile2D<T>,
+) -> TemporalRasterAggregationTileAccu<T>
+where
+    T: Pixel,
+{
+    let mut next_accu = tile;
+    next_accu.time = acc.accu_tile.time;
+
+    TemporalRasterAggregationTileAccu {
+        accu_tile: next_accu,
+        initial_state: false,
+    }
+}
+
+pub fn last_tile_fold_future<T>(
+    accu: TemporalRasterAggregationTileAccu<T>,
+    tile: RasterTile2D<T>,
+) -> impl Future<Output = Result<TemporalRasterAggregationTileAccu<T>>>
+where
+    T: Pixel,
+{
+    tokio::task::spawn_blocking(|| last_tile_fold_fn(accu, tile)).then(async move |x| match x {
+        Ok(r) => Ok(r),
+        Err(e) => Err(e.into()),
+    })
+}
+
+#[derive(Debug, Clone)]
+pub struct TemporalRasterAggregationTileAccu<T> {
```

Comment:
why do you need this? can't you just use a `Option<RasterTile2D>` as accumulator and start with None?

Reply:
well i could implement the trait for option i guess.